### PR TITLE
Minor refactor of d2cof

### PR DESCRIPTION
**Pull request by gravestench**
* Changed `Load` to `Unmarshal`
* made `Marshal` and `Unmarshal` into methods of `COF`
* added `New` function which creates a new, empty COF instance
* added helper functions `Marshal` and `Unmarshal` (separate from the COF methods, these are package-level functions)
* Changed `StreamReader.ReadBytes` to account for edge case of reading 0
bytes (this was returning an error when it should not have)
* added really simple unit tests for COF

### CODE REVIEW

File: d2common/d2fileformats/d2cof/cof.go [hunk 1]
```diff
@@ -33,12 +34,41 @@ const (
...
+		CompositeLayers:    make(map[d2enum.CompositeType]int),
+		AnimationFrames:    make([]d2enum.AnimationFrame, 0),
+		Priority:           make([][][]d2enum.CompositeType, 0),
+	}
+}
+
+// Marshal a COF to a new byte slice
+func Marshal(c *COF) []byte {
+	return c.Marshal()
+}
```
Comment (gucio321) [inline d2common/d2fileformats/d2cof/cof.go:56]: Is it necessary?
In my opinion we'll easier to use
```go
cof.Marshal()
```
instead of
```golang
d2cof.Marshal(cof)
```

**Review by essial — APPROVED** [review]
